### ISSUE TITLE
Capture Feature does not capture any events and displays unhandled exceptions

### ISSUE THREAD

**Issue by strassi** [rabbitstack/fibratus#239]
Hi,

in can't figure out what I'm doing wrong by using the capture feature of fibratus. 

**Setup:**
```
PS C:\Windows\Temp> Get-ComputerInfo
WindowsProductName                                      : Windows 10 Pro
WindowsBuildLabEx                                       : 22621.1.amd64fre.ni_release.220506-1250
OSDisplayVersion                                        : 22H2
WindowsVersion                                          : 2009
BiosCurrentLanguage                                     : en|US|iso8859-1,0
CsProcessors                                            : {12th Gen Intel(R) Core(TM) i7-1260P}
OsTotalVisibleMemorySize                                : 16478712
....
PS C:\Windows\Temp> fibratus.exe version
┌─────────────┬─────────────────────┐
│ Version     │ 2.0.0               │
│ Commit      │ 2268bda4            │
│ Build date  │ 01-09-2023.17:40:30 │
├─────────────┼─────────────────────┤
│ Go compiler │ go1.19.12           │
└─────────────┴─────────────────────┘
PS C:\Windows\Temp> whoami /priv

PRIVILEGES INFORMATION
----------------------

Privilege Name                            Description                                                        State
========================================= ================================================================== =======
SeIncreaseQuotaPrivilege                  Adjust memory quotas for a process                                 Enabled
SeSecurityPrivilege                       Manage auditing and security log                                   Enabled
SeTakeOwnershipPrivilege                  Take ownership of files or other objects                           Enabled
SeLoadDriverPrivilege                     Load and unload device drivers                                     Enabled
SeSystemProfilePrivilege                  Profile system performance                                         Enabled
SeSystemtimePrivilege                     Change the system time                                             Enabled
SeProfileSingleProcessPrivilege           Profile single process                                             Enabled
SeIncreaseBasePriorityPrivilege           Increase scheduling priority                                       Enabled
SeCreatePagefilePrivilege                 Create a pagefile                                                  Enabled
SeBackupPrivilege                         Back up files and directories                                      Enabled
SeRestorePrivilege                        Restore files and directories                                      Enabled
SeShutdownPrivilege                       Shut down the system                                               Enabled
SeDebugPrivilege                          Debug programs                                                     Enabled
SeSystemEnvironmentPrivilege              Modify firmware environment values                                 Enabled
SeChangeNotifyPrivilege                   Bypass traverse checking                                           Enabled
SeRemoteShutdownPrivilege                 Force shutdown from a remote system                                Enabled
SeUndockPrivilege                         Remove computer from docking station                               Enabled
SeManageVolumePrivilege                   Perform volume maintenance tasks                                   Enabled
SeImpersonatePrivilege                    Impersonate a client after authentication                          Enabled
SeCreateGlobalPrivilege                   Create global objects                                              Enabled
SeIncreaseWorkingSetPrivilege             Increase a process working set                                     Enabled
SeTimeZonePrivilege                       Change the time zone                                               Enabled
SeCreateSymbolicLinkPrivilege             Create symbolic links                                              Enabled
SeDelegateSessionUserImpersonatePrivilege Obtain an impersonation token for another user in the same session Enabled
```

**Actual behavior:**

```
fibratus capture "kevt.category = 'file'" -o fs-events --config-file 'C:\Program Files\Fibratus\Config\fibratus.yml' --logging.log-stdout --logging.level debug

{"level":"debug","msg":"starting kernel trace with \"DiskFileIO, FileIO, DLL, TCPIP, Process, Registry, Thread, VaMap, VirtualAlloc\" event flags","time":"2024-01-16T18:00:34+01:00"}
{"level":"debug","msg":"starting trace [NT Kernel Logger]","time":"2024-01-16T18:00:34+01:00"}
{"level":"debug","msg":"starting trace [Kernel Audit API Calls Logger]","time":"2024-01-16T18:00:34+01:00"}
{"level":"debug","msg":"starting trace [DNS Client Logger]","time":"2024-01-16T18:00:34+01:00"}
{"level":"info","msg":"starting [NT Kernel Logger] trace processing","time":"2024-01-16T18:00:34+01:00"}
{"level":"info","msg":"starting [Kernel Audit API Calls Logger] trace processing","time":"2024-01-16T18:00:34+01:00"}
{"level":"info","msg":"starting [DNS Client Logger] trace processing","time":"2024-01-16T18:00:34+01:00"}
> Capturing ⠦ {"level":"warning","msg":"maximum handle table size reached for 1220 pid. Shrinking table size from 1285 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 1360 pid. Shrinking table size from 1428 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 1404 pid. Shrinking table size from 978 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
> Capturing ⠧ {"level":"warning","msg":"maximum handle table size reached for 3568 pid. Shrinking table size from 1372 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
> Capturing ⠇ {"level":"warning","msg":"maximum handle table size reached for 12480 pid. Shrinking table size from 2042 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
> Capturing ⠏ {"level":"warning","msg":"maximum handle table size reached for 10604 pid. Shrinking table size from 5337 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 6344 pid. Shrinking table size from 1790 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 14148 pid. Shrinking table size from 970 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 1764 pid. Shrinking table size from 814 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 9928 pid. Shrinking table size from 961 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 6136 pid. Shrinking table size from 816 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
> Capturing ⠋ {"level":"warning","msg":"maximum handle table size reached for 15324 pid. Shrinking table size from 1209 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 16032 pid. Shrinking table size from 991 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 8328 pid. Shrinking table size from 823 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 15328 pid. Shrinking table size from 843 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 4644 pid. Shrinking table size from 2118 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 1748 pid. Shrinking table size from 1030 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 12480 pid. Shrinking table size from 2042 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 3032 pid. Shrinking table size from 1460 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 13908 pid. Shrinking table size from 1175 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
> Capturing ⠙ {"level":"warning","msg":"maximum handle table size reached for 16016 pid. Shrinking table size from 904 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 10604 pid. Shrinking table size from 5337 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 7960 pid. Shrinking table size from 809 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 6344 pid. Shrinking table size from 1790 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 14148 pid. Shrinking table size from 970 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 1764 pid. Shrinking table size from 814 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
> Capturing ⠹ {"level":"warning","msg":"maximum handle table size reached for 9928 pid. Shrinking table size from 961 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 6136 pid. Shrinking table size from 816 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 15324 pid. Shrinking table size from 1209 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 16032 pid. Shrinking table size from 991 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 8328 pid. Shrinking table size from 823 to 800 handles","time":"2024-01-16T18:00:36+01:00"}
> Capturing ⠸ {"level":"warning","msg":"maximum handle table size reached for 15328 pid. Shrinking table size from 843 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 4644 pid. Shrinking table size from 2118 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 1748 pid. Shrinking table size from 1030 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 3032 pid. Shrinking table size from 1460 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
> Capturing ⠼ {"level":"warning","msg":"maximum handle table size reached for 13908 pid. Shrinking table size from 1175 to 800 handles","time":"2024-01-16T18:00:37+01:00"}                                      {"level":"warning","msg":"maximum handle table size reached for 16016 pid. Shrinking table size from 904 to 800 handles","time":"2024-01-16T18:00:37+01:00"}                                                     {"level":"warning","msg":"maximum handle table size reached for 7960 pid. Shrinking table size from 809 to 800 handles","time":"2024-01-16T18:00:37+01:00"}                                                      > Capturing ⠦ {"level":"warning","msg":"maximum handle table size reached for 1220 pid. Shrinking table size from 1285 to 800 handles","time":"2024-01-16T18:00:37+01:00"}                                       {"level":"warning","msg":"maximum handle table size reached for 1360 pid. Shrinking table size from 1428 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 1404 pid. Shrinking table size from 978 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
> Capturing ⠧ {"level":"warning","msg":"maximum handle table size reached for 3568 pid. Shrinking table size from 1372 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
> Capturing ⠇ {"level":"warning","msg":"maximum handle table size reached for 12480 pid. Shrinking table size from 2042 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 10604 pid. Shrinking table size from 5337 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 6344 pid. Shrinking table size from 1790 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
> Capturing ⠏ {"level":"warning","msg":"maximum handle table size reached for 14148 pid. Shrinking table size from 970 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 1764 pid. Shrinking table size from 814 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 9928 pid. Shrinking table size from 961 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 6136 pid. Shrinking table size from 816 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 15324 pid. Shrinking table size from 1209 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 16032 pid. Shrinking table size from 991 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 8328 pid. Shrinking table size from 823 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 15328 pid. Shrinking table size from 843 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 4644 pid. Shrinking table size from 2118 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 1748 pid. Shrinking table size from 1030 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
> Capturing ⠋ {"level":"warning","msg":"maximum handle table size reached for 3032 pid. Shrinking table size from 1460 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 13908 pid. Shrinking table size from 1175 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 16016 pid. Shrinking table size from 904 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
{"level":"warning","msg":"maximum handle table size reached for 7960 pid. Shrinking table size from 809 to 800 handles","time":"2024-01-16T18:00:37+01:00"}
> Capturing ⠧ ERROR msg=unhandled exception when parsing data directory Import, reason: runtime error: slice bounds out of range [4294967295:97632]
> Capturing ⠴ ERROR msg=unhandled exception when parsing data directory Import, reason: runtime error: slice bounds out of range [4294967295:578384]
ERROR msg=unhandled exception when parsing data directory Import, reason: runtime error: slice bounds out of range [4294967295:109440]
> Capturing ⠋ {"level":"info","msg":"got signal \"interrupt\", shutting down...","time":"2024-01-16T18:02:09+01:00"}
{"level":"info","msg":"stopping [NT Kernel Logger] trace processing","time":"2024-01-16T18:02:10+01:00"}
{"level":"info","msg":"stopping [Kernel Audit API Calls Logger] trace processing","time":"2024-01-16T18:02:10+01:00"}
{"level":"info","msg":"stopping [DNS Client Logger] trace processing","time":"2024-01-16T18:02:10+01:00"}
{"level":"warning","msg":"CloseTrace: invalid argument","time":"2024-01-16T18:02:10+01:00"}
{"level":"warning","msg":"CloseTrace: invalid argument","time":"2024-01-16T18:02:10+01:00"}
{"level":"warning","msg":"CloseTrace: invalid argument","time":"2024-01-16T18:02:10+01:00"}
┌────────────────────────────────────┐
│ Capture Statistics                 │
├───────────────────┬────────────────┤
│ File              │ fs-events.kcap │
├───────────────────┼────────────────┤
│ Events written    │ 0              │
│ Bytes written     │ 0              │
│ Processes written │ 0              │
│ Handles written   │ 0              │
├───────────────────┼────────────────┤
│ Capture size      │ 37 B           │
└───────────────────┴────────────────┘
PS C:\Windows\Temp> Get-Item .\fs-events.kcap


    Directory: C:\Windows\Temp


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----         1/16/2024   6:02 PM             40 fs-events.kcap


PS C:\Windows\Temp> Get-Content .\fs-events.kcap
(µ/ýXàsutarbif
```

**Expected Behavior:**

File Event in the file .\fs-events.kcap

**Additional Information:**

I do not know if the line `ERROR msg=unhandled exception when parsing data directory Import, reason: runtime error: slice bounds out of range [4294967295:109440]` might cause no events written to the capture file.

I tried to capture events without any filters, but it results in the same issue.

If I run `fibratus.exe run` the output displays a lot of events. So the events are being seen and fibratus seems to work.

I tried to run the commands from powershell (elevated) and cmd.exe (elevated).

The log files do not provide additional information besides the same lines as the stdout (as expected with the command line switch --logging.log-stdout).

Thank you for your time on this issue in advance.

**Comment by rabbitstack** [comment]
Hi @strassi ,

I appreciate the detailed instructions on how to reproduce the bug. Does this only manifest with the latest version?

I'll look into this ASAP and let you know my findings.

**Comment by rabbitstack** [comment]
Thanks for confirming and testing with the previous release. It is pretty evident, we have a regression impacting the version 2.0.0. I'm going to fix this in the next release (2.2.0).

**Comment by rabbitstack** [comment]
Hi @strassi,

The PR with a bug fix was just submitted. Again, thanks for catching this.
